tests/wpt/web-platform-tests/resources/testharness.js

```diff
@@ -921,21 +921,23 @@ policies and contribution forms [3].
 
              var p;
              for (p in actual) {
-                 assert(expected.hasOwnProperty(p), "assert_object_equals", description,
+                 assert(p in expected, "assert_object_equals", description,
```

Comment:
No, this is fine syntax. However, there's no way you're going to just randomly specialize a generic function for this one test. Instead, stop using `assert_object_equals` ([it's going to be removed anyway](https://github.com/w3c/web-platform-tests/issues/2033)).